wagtail/admin/views/pages/choose_parent.py

```diff
@@ -0,0 +1,85 @@
+from django.contrib.admin.utils import quote
+from django.contrib.contenttypes.models import ContentType
+from django.shortcuts import redirect
+from django.views import View
+
+from wagtail.admin.forms.pages import ParentChooserForm
+from wagtail.admin.views.generic.base import WagtailAdminTemplateMixin
+from wagtail.models import Page
+
+
+class ChooseParentView(WagtailAdminTemplateMixin, View):
+    template_name = "wagtailadmin/pages/choose_parent.html"
+    model = Page
+
+    def get_valid_parent_pages(self, user):
+        """
+        Identifies possible parent pages for the current user by first looking
+        at allowed_parent_page_models() on self.model to limit options to the
+        correct type of page, then checking permissions on those individual
+        pages to make sure we have permission to add a subpage to it.
+        """
+        # Get queryset of pages where this page type can be added
+        allowed_parent_page_content_types = list(
+            ContentType.objects.get_for_models(
+                *self.model.allowed_parent_page_models()
+            ).values()
+        )
+        allowed_parent_pages = Page.objects.filter(
+            content_type__in=allowed_parent_page_content_types
+        )
+
+        # Get queryset of pages where the user has permission to add subpages
+        if user.is_superuser:
+            pages_where_user_can_add = Page.objects.all()
+        else:
+            pages_where_user_can_add = Page.objects.none()
+
+            from wagtail.permission_policies.pages import PagePermissionPolicy
```

Comment:
We should use the singular instance of the policy as of #11267

```suggestion
            from wagtail.permissions import page_permission_policy
```

and move this to the top-level imports